rr_rviz_plugins/src/ChassisPanel.cpp

```diff
@@ -0,0 +1,198 @@
+    painter.setPen(QColor(255, 255, 255, 255));
+    painter.setBrush(QColor(255, 255, 255, 255));
+    painter.drawRect(rect);
+    painter.setBrush(Qt::black);
+    painter.setPen(Qt::black);
+    int x_org = x_pos + width / 2;
+    int y_org = y_pos + height / 2;
+    // The size of the arrow
+    int arr_size = 9;  // this is actually  half the length of the arrow
+    // makes it so that zero points north instead of east
+    hding = hding + M_PI / 2;
+    y_comp = sin(hding) * (float)arr_size;
+    x_comp = cos(hding) * (float)arr_size;
+
+    QLineF arrow_shaft = QLineF(x_org - x_comp, y_org + y_comp, x_org + x_comp, y_org - y_comp);
+    painter.drawLine(arrow_shaft);
+
+    // For the arrowhead
+    painter.setPen(Qt::red);
+    painter.setBrush(Qt::red);
+    QLineF arrow_tip = QLineF(x_org + x_comp, y_org - y_comp, x_org + .5 * x_comp, y_org - .5 * y_comp);
+    painter.drawLine(arrow_tip);
+
+    // For the status of recording_bag
+    x_pos = 137;
+    y_pos = 95;
+    width = 40;
+    height = 15;
+    if (received_signal && record_bag) {
+        painter.setBrush(Qt::green);
+        painter.setPen(Qt::green);
+        painter.drawRect(x_pos, y_pos, width, height);
+    } else if (received_signal) {
+        painter.setBrush(Qt::red);
+        painter.setPen(Qt::red);
+        painter.drawRect(x_pos, y_pos, width, height);
+    }
+
+    // for the status of estop
+    x_pos = 52;
+    y_pos = 122;
+    width = 23;
+    if (received_signal && estop) {
+        painter.setBrush(Qt::red);
+        painter.setPen(Qt::red);
+        painter.drawRect(x_pos, y_pos, width, height);
+    } else if (received_signal) {
+        painter.setBrush(Qt::green);
+        painter.setPen(Qt::green);
+        painter.drawRect(x_pos, y_pos, width, height);
+    }
+    QWidget::paintEvent(e);
+}
+
+void ChassisPanel::chassisStateCallback(const rr_msgs::chassis_state msg) {
+    // for timerCallback
+    chassisUpdateTime = ros::Time::now();
+    received_signal = true;
+
+    // global variables
+    cur_speed = msg.speed_mps;
+    hding = msg.steer_rad;
+    record_bag = msg.record_bag;
+    estop = msg.estop_on;
+
+    // label updates
+    auto speed = spacing.toStdString() + std::to_string(msg.speed_mps) + " m/s";
+    auto heading = spacing.toStdString() + std::to_string(msg.steer_rad) + " rad";
+    auto message = msg.state;
+    speed_label->setText(speed.c_str());
+    heading_label->setText(heading.c_str());
+    msg_label->setText(message.c_str());
+
+    if (msg.record_bag) {
+        rcding_bag_label->setText("Recording Bag File:  true");
+        // rcding_bag_label->setStyleSheet("QLabel { background-color : green}");
+    } else {
+        rcding_bag_label->setText("Recording Bag File:  false");
+        // rcding_bag_label->setStyleSheet("QLabel { background-color : red}");
+    }
+    if (msg.estop_on) {
+        estop_label->setText("Estop: on");
+    } else {
+        estop_label->setText("Estop: off");
+    }
```

Comment:
You probably can convert this to two lines just like how you did for the other ones.